sdk/dotnet/Pulumi/Serialization/BoolOutputCompletionSource.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright 2016-2018, Pulumi Corporation
+
+#nullable enable
+
+namespace Pulumi.Rpc
+{
+    public sealed class BoolOutputCompletionSource : ProtobufOutputCompletionSource<bool>

```

Comment:
These types in the `Pulumi.Rpc` namespace need to be public?

Reply:
these will be.  They'll be needed by our different provider packages (i.e. Pulumi.Aws/Pulumi.Azure) to mark their Output properties so that hte core layer can deserialize results into them.  Not certain where they shoudl live.  I don't want them in the root Pulumi namespace, but they do need to be public somewhere.